pkg/db/migration.go

```diff
@@ -109,13 +110,28 @@ func getMigrate(params params.Database) (*migrate.Migrate, error) {
 	if connectionString == "" {
 		connectionString = "postgres://:/"
 	}
-	m, err := migrate.NewWithSourceInstance("httpfs", src, connectionString)
+	m, err := tryNewWithSourceInstance("httpfs", src, connectionString)
 	if err != nil {
 		return nil, err
 	}
 	return m, nil
 }
 
+func tryNewWithSourceInstance(sourceName string, sourceInstance source.Driver, databaseURL string) (*migrate.Migrate, error) {
+	strategy := params.DatabaseRetryStrategy
+	var err error
+	for a := retry.Start(strategy, nil); a.Next(); {
+		m, err := migrate.NewWithSourceInstance(sourceName, sourceInstance, databaseURL)
+		if err == nil {
```

Comment:
Without checking the actual error for the cause of the failure, how can we know that the failure was because of a connection and not a failure to perform the actual action?

Reply:
Also not sure if it's safe to try to migrate twice, it might failed half way through..

Reply:
`migrate.NewWithSourceInstance` doesn't actually perform the migration, it only gets the Migrate instance